main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
I generally agree with @jvican. I would never copy-paste `ProjectRef(uri("file:/xxx/helloworld/"),"root")/Compile/scalacOptions` inside my build. I would never *write* that in my build either.

There are two things "we" want here:

* Copy-past'ability
* No loss of information whatsoever

IMO those goals are at odds. You may choose one *or* the other. Both is unreasonable. *Yes*, *if I did* copy-paste `ProjectRef(uri("file:/xxx/helloworld/"),"root")/Compile/scalacOptions` in my build, it would *work*, but it would destroy the readability and the cleanliness of my build code. I would never do that, and I certainly don't want beginners to do it because we told them they could copy-paste from the output of some task. It will confuse even more beginners, will produce builds that are even more obscure and unreadable to the uninitiated *and* the expert alike. There is no advantage whatsoever.